src/tools/pub/automated-publishing.md

```diff
@@ -0,0 +1,719 @@
+check in generated or built artifacts into your repository, you
+can create a custom workflow. This workflow could resemble the
+following YAML markup:
+
+```yaml
+# .github/workflows/publish.yml
+name: Publish to pub.dev
+
+on:
+  push:
+    tags:
+    - 'v[0-9]+.[0-9]+.[0-9]+*' # tag pattern on pub.dev: 'v{{version}}'
+
+# Publish using custom workflow
+jobs:
+  publish:
+    permissions:
+      id-token: write # This is required for authentication using OIDC
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dart-lang/setup-dart
+      - name: Install dependencies
+        run: dart pub get
+      # Here you can insert custom steps you need
+      # - run: dart tool/generate-code.dart
+      - name: Publish
+        run: dart pub publish --force
+```
+
+The workflow authenticates to `pub.dev` using a temporary [token][1].
+Create and configure this token in the `dart-lang/setup-dart` `step` in
+the `publish.yaml` file.
+To publish to `pub.dev`, subsequent steps can run
+`dart pub publish --force`.
+
+{{site.alert.note}}
+  At this point in this guide, anyone with push access to your
+  repository can publish new versions of the package.
+  Consider using [tag protection rules][sec-gh-tag-protection] or
+  [GitHub deployment Environments][sec-gh-environment] to limit who can
+  publish.
+{{site.alert.end}}
+
+[sec-gh-tag-protection]: #hardening-security-with-tag-protection-rules-on-github
+[sec-gh-environment]: #hardening-security-with-github-deployment-environments
+
+
+## Trigger automated publishing using Github Actions
+
+After you have completed the preceding activities, you can publish a
+new version of your package.
+
+1. Verify your package's version.
+
+    ```console
+    $ cat pubspec.yaml
+    package: my_package_name
+    version: 1.2.3            # must match the version number used in the git tag
+    environment:
+      sdk: ^2.18.0
+    ...
+    ```
+
+1. Tag the latest commit with the matching `tag-pattern`.
+
+    ```console
+    $ git tag v1.2.3          # assuming my tag pattern is: 'v{{version}}'
+    ```
+
+1. Push the tag to GitHub.
+
+    ```console
+    $ git push origin v1.2.3  # triggers the action that publishes my package.
+    ```
+
+1. Review the GitHub Action workflow logs at
+   `https://github.com/<organization>/<repository>/actions`.
+
+If you don't like using the `git` command to create tags, you can
+create _releases_ on GitHub.
+To learn more, check out [managing releases in a repository][4] in the
+GitHub documentation.
+
+
+## Troubleshoot issues with GitHub Actions
+
+* If the Action didn't trigger, check that the pattern configured in
+  `.github/workflows/publish.yml` matches the pushed _git tag_.
+* If the Action failed, the GitHub Action logs may contain clues as to
+  why it failed. If the log offered no clues, check that you set the
+  proper `organization/repository` and `tag-pattern` on `pub.dev`.
```

Comment:
I think we can just leave this as a paragraph, it's just two lines, suggesting that you do the obvious, if things don't work. This is not a step by step troubleshooting guide.

We're not making such a guide, we're documeting how things work, and hoping users can put 2 and 2 together.